storage/storage_image.go

```diff
@@ -759,6 +780,12 @@ func (s *storageImageDestination) Commit(ctx context.Context, unparsedToplevel t
 	if intendedID == "" {
 		intendedID = s.computeID(man)
 	}
+	for _, lid := range s.blobLayerIDs {
+		// If we use custom layer IDs for creating an image, we should include this
+		// information as well. This prevents the conflict of between image ID of an image
+		// with custom ID and the one without custom ID.
+		intendedID = digest.SHA256.FromString(fmt.Sprintf("%s %s", intendedID, lid)).Encoded()
```

Comment:
What is a “custom ID”?

Why does the presence of additional-layer-store layers matter at all? It’s all the same image, so it should be deduplicated, shouldn’t it?

Reply:
> What is a “custom ID”?

Now we don't have it.